src/main/scala/cromwell/engine/workflow/WorkflowActor.scala

```diff
@@ -376,35 +375,44 @@ case class WorkflowActor(workflow: WorkflowDescriptor, backend: Backend)
       stay() using data.copy(startMode = Option(startMode))
   }
 
-  private def handleCallRetried(callKey: OutputKey,
-                                  retryStatus: ExecutionStatus,
-                                  returnCode: Option[Int],
-                                  failure: Throwable,
-                                  message: CallFailedRetryable,
-                                  data: WorkflowData,
-                                  events: Seq[ExecutionEventEntry]) = {
-    val currentSender = sender()
+  private def retryCall(callKey: ExecutionStoreKey,
+                        data: WorkflowData) = {
+
     val keyClone = callKey.retryClone
-    executionStore += keyClone -> ExecutionStatus.NotStarted
 
-    val completionWork = for {
-      _ <- globalDataAccess.setExecutionEvents(workflow.id, callKey.scope.fullyQualifiedName, callKey.index, callKey.attempt, events)
-      _ <- globalDataAccess.insertCalls(workflow.id, Seq(keyClone), backend)
-    } yield ()
+    val insertCopy = globalDataAccess.insertCalls(workflow.id, Seq(keyClone), backend)
 
-    completionWork onComplete {
-      case Failure(e) =>
-        logger.error(s"Preemption work failed for call ${callKey.tag}! " + e.getMessage, e)
-        currentSender ! CallActor.Ack(message)
+    insertCopy map { _ => PersistenceSucceeded(keyClone, ExecutionStatus.NotStarted) } pipeTo self
+
+    insertCopy onFailure {
+      case e =>
+        logger.error(s"Failed to clone ${callKey.tag} for retry attempt " + e.getMessage, e)
         self ! CheckForWorkflowComplete
```

Comment:
Guessing here on the reason it was put there (and not saying the logic is actually correct) but I'd bet that the thought was that if that led to the workflow failing you'd want to detect it